data/grub_auth/create_custom_grub.sh

```diff
@@ -3,47 +3,62 @@
 set -euo pipefail
 
 ROOT_DISK=`df -T | grep \/$ | awk '{print $1}'`
-PARTNO=`echo ${ROOT_DISK:0-1}`
 FS=`df -T | grep \/$ | awk '{print $2}'`
+UUID=`blkid | grep $ROOT_DISK | awk -F \" '{print $2}'`
+ARCH=`uname -m`
+KERNEL=""
 SUP_USER="$1"
 SUP_PASSWD="$2"
 MAINT_USER="$3"
 MAINT_PASSWD="$4"
+SUPER_PDKDF2=`cat /tmp/sup_passwd_hash | grep PBKDF2 | awk -F "password is " '{print \$2}'`
+MAINT_PDKDF2=`cat /tmp/maint_passwd_hash | grep PBKDF2 | awk -F "password is " '{print \$2}'`
 
 : > /etc/grub.d/10_linux
 
 if [ $FS = "ext4" ] || [ $FS = "ext3" ]; then
 FS=ext2
 fi
 
+if [ $ARCH = "x86_64" ]; then
+KERNEL="vmlinuz"
+elif [ $ARCH = "aarch64" ]; then
+KERNEL="Image"
+elif [ $ARCH = "s390x" ]; then
+KERNEL="image"
+else
+KERNEL="vmlinux"
+fi
+
 source  /etc/default/grub
 
 (
 
 cat <<EOF
 set superusers=$SUP_USER
-password $SUP_USER $SUP_PASSWD
-password $MAINT_USER $MAINT_PASSWD
+password_pbkdf2 $SUP_USER $SUPER_PDKDF2
```

Comment:
could you pls add some comment for this usage?

Reply:
sure